test/e2e/deployment.go

```diff
@@ -1319,3 +1322,85 @@ func testOverlappingDeployment(f *framework.Framework) {
 	err = framework.WaitForDeploymentRevisionAndImage(c, ns, deployOverlapping.Name, "2", redisImage)
 	Expect(err).NotTo(HaveOccurred(), "The second deployment failed to update to revision 2")
 }
+
+func testFailedDeployment(f *framework.Framework) {
```

Comment:
Can we add a test that stress tests the pause / unpause and also tries to create a "fast failure" scenario?  I'm thinking like the openshift scenario suite that stresses the iterative execution of the deployer.